spec/controllers/spree/amazon_controller_spec.rb

```diff
@@ -161,7 +161,15 @@ def stub_amazon_order(address: build_amazon_address, email: 'jordan.brough@examp
       stub_amazon_order
       set_current_order(order)
 
-      expect_any_instance_of(SpreeAmazon::Order).to receive(:save_total)
+      expect_any_instance_of(SpreeAmazon::Order).to(
```

Comment:
Could we rewrite this example using spies so we have 3 clear steps? Setup, execution and validation?

Something like:
```ruby
allow_any_instance_of(SpreeAmazon::Order).to receive(:set_order_reference_details)
allow_any_instance_of(SpreeAmazon::Order).to receive(:confirm)

spree_post :confirm

expect_any_instance_of(SpreeAmazon::Order).to have_received(:set_order_reference_details).with(...)
expect_any_instance_of(SpreeAmazon::Order).to have_received(:confirm)
```